kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/avro/AvroRecordWriterProvider.java

```diff
@@ -76,7 +76,7 @@ public void write(SinkRecord record) {
             writer.setCodec(CodecFactory.fromString(conf.getAvroCodec()));
             writer.create(avroSchema, s3out);
           } catch (IOException e) {
-            throw new ConnectException(e);
+            throw S3ErrorUtils.maybeRetriableConnectException(e);
```

Comment:
nit: I think that the `maybeRetriableConnectException` being a factory method for creating an exception is slightly misleading. What do you think about making the method construct and throw the correct exception and just calling it something like `throwConnectException` ?

Reply:
Well, I did the change, and it looks fine by me, except one place:
In `JsonRecordWriterProvider.java:126` I had to add a dummy `return null` statement, because otherwise compiler complains about "missing return statement". Which is not a big deal I think. What do you think?